src/cascadia/TerminalSettingsModel/GlobalAppSettings.cpp

```diff
@@ -161,6 +161,7 @@ void GlobalAppSettings::DefaultProfile(const winrt::guid& defaultProfile) noexce
 {
     _validDefaultProfile = true;
     _defaultProfile = defaultProfile;
+    _UnparsedDefaultProfile = Utils::GuidToString(defaultProfile);

```

Comment:
What happened here?

Reply:
Past Carlos messed up. This is in the `GlobalAppSettings::ToJson()`:
```c++
JsonUtils::SetValueForKey(json, DefaultProfileKey,              _UnparsedDefaultProfile);
```

SUI updates TSM via the getter/setter for `DefaultProfile`. When we call the setter, we need to update `_UnparsedDefaultProfile` so that we serialize it in `ToJson()`.

Side note: this was the first setting I tested and I thought I broke serialization haha. Turns out, this was the only setting that had an issue like this because we do this weird validation stuff. profile.colorScheme does something kinda similar but is hooked up in another PR.

Reply:
okay, that makes sense, thanks!